apps/src/aichat/redux/aichatRedux.ts

```diff
@@ -209,13 +211,24 @@ export const onSaveComplete =
 
     changedProperties.forEach(property => {
       const typedProperty = property as keyof AiCustomizations;
+      const modelUpdate = {
+        id: getNewMessageId(),
+        updatedField: typedProperty,
+        updatedValue: currentAiCustomizations[typedProperty],
+        timestamp: Date.now(),
+      };
+      dispatch(addModelUpdate(modelUpdate));
+      const {updatedField, updatedValue} = modelUpdate;
+      // Only log updated value for temperature and selected model id - free text values are not logged.
+      const updatedValueToLog =
+        updatedField === 'temperature' || updatedField === 'selectedModelId'
+          ? updatedValue
+          : 'N/A';
       dispatch(
-        addModelUpdate({
-          id: getNewMessageId(),
-          updatedField: typedProperty,
-          updatedValue: currentAiCustomizations[typedProperty],
-          timestamp: Date.now(),
-        })
+        logChatEvent({
```

Comment:
I think it might be good to reorganize this a little bit. We could have a single thunk that all events are routed through, which logs the event, and adds to the current set of events if necessary. That way we're not at risk of forgetting to call `logChatEvent` in every place that we also add a chat event to the window. For example:

```
const addEvent = <T extends ChatEvent>(event: T) =>
  (dispatch: AppDispatch, getState: () => RootState) => {
    // Log all events except for notifications that are not included in history (i.e. non-profanity save errors).
    if (
      !isNotification(event) ||
      (isNotification(event) && event.includeInHistory) // Let's assume we add some `includeInHistory` flag to Notification, so we don't need to check the profanity message string directly
    ) {
      const {progress, lab} = getState();
      const aichatContext: AichatContext = {
        currentLevelId: parseInt(progress.currentLevelId || ''),
        scriptId: progress.scriptId,
        channelId: lab.channel?.id,
      };
      ChatEventLogger.getInstance().logChatEvent(event, aichatContext);
    }

    // Add to current chat window if visible.
    if (!event.hideForParticipants) {
      dispatch(addEventToCurrentChat(event)); // addEventToCurrentChat (can be named whatever) would be a new redux action to replace addChatMessage/addModelUpdate/addNotification, that adds the event to "chatEventsCurrent"
    }
  };
```

Then, we can replace all usages of `addChatMessage`, `addModelUpdate`, and `addNotification` with simply `addEvent`. `addChatMessage` and `addModelUpdate` are only used internally in this file as it is; `addNotification` is currently used externally only for the reset model notification, which we could just create a dedicated thunk for

Reply:
Nice idea! Updated. Don't think I need `addNotification` as can use `addChatEvent` for the reset model notification also. Pushed the updates and tested locally and confirmed logging is as expected. Lmk what you think. Thanks!